broker/cloud_run/lsst/classify_snn/main.py

```diff
@@ -0,0 +1,170 @@
+PROJECT_ID = os.getenv("GCP_PROJECT")
+TESTID = os.getenv("TESTID")
+SURVEY = os.getenv("SURVEY")
+
+# provenance variables
+MODULE_NAME = "SuperNNova"
+MODULE_VERSION = 0.1
+
+# classifier variables
+model_dir_name = "ZTF_DMAM_V19_NoC_SNIa_vs_CC_forFink"
+model_file_name = (
+    "vanilla_S_0_CLF_2_R_none_photometry_DF_1.0_N_global_lstm_32x2_0.05_128_True_mean.pt"
+)
+MODEL_PATH = Path(__file__).resolve().parent / model_dir_name / model_file_name
+
+# Variables for incoming data
+# A url route is used in setup.sh when the trigger subscription is created.
+# It is possible to define multiple routes in a single module and trigger them using different subscriptions.
+ROUTE_RUN = "/"  # HTTP route that will trigger run(). Must match setup.sh
+
+# Variables for outgoing data
+HTTP_204 = 204  # HTTP code: Success
+HTTP_400 = 400  # HTTP code: Bad Request
+
+# GCP resources used in this module
+# pittgoogle will construct the full resource names from the MODULE_NAME, SURVEY, and TESTID
+TOPIC = pittgoogle.Topic.from_cloud(
+    MODULE_NAME, survey=SURVEY, testid=TESTID, projectid=PROJECT_ID
+)
+TOPIC_BIGQUERY_IMPORT_SUPERNNOVA = pittgoogle.Topic.from_cloud(
+    "bigquery-import-SuperNNova", survey=SURVEY, testid=TESTID, projectid=PROJECT_ID
+)
+
+TOPIC_BIGQUERY_IMPORT_CLASSIFICATIONS = pittgoogle.Topic.from_cloud(
+    "bigquery-import-classifications", survey=SURVEY, testid=TESTID, projectid=PROJECT_ID
+)
+
+app = flask.Flask(__name__)
+
+
+@app.route(ROUTE_RUN, methods=["POST"])
+def run():
+    """Classify the alert; publish and store results.
+
+    This module is intended to be deployed as a Cloud Run service. It will operate as an HTTP endpoint
+    triggered by Pub/Sub messages. This function will be called once for every message sent to this route.
+    It should accept the incoming HTTP request and return a response.
+
+    Returns
+    -------
+    response : tuple(str, int)
+        Tuple containing the response body (string) and HTTP status code (int). Flask will convert the
+        tuple into a proper HTTP response. Note that the response is a status message for the web server
+        and should not contain the classification results.
+    """
+    # extract the envelope from the request that triggered the endpoint
+    # this contains a single Pub/Sub message with the alert to be processed
+    envelope = flask.request.get_json()
+
+    # unpack the alert. raises a `BadRequest` if the envelope does not contain a valid message
+    try:
+        alert = pittgoogle.Alert.from_cloud_run(envelope, "lsst")
+    except pittgoogle.exceptions.BadRequest as exc:
+        return str(exc), HTTP_400
+
+    # classify
+    snn_dict = _classify(alert)
+
+    # prepare data for publishing
+    classifier_summary = _classification_summary(snn_dict)
+    snn_alert = _create_outgoing_alert(alert, snn_dict)
+    snn_results = pittgoogle.Alert.from_dict(payload=snn_dict)
+
+    # publish
+    TOPIC.publish(snn_alert, serializer="avro")
```

Comment:
Does this actually work with the avro serializer? I think it's going to try to use LSST's schema and that won't match snn_alert.

Reply:
Haven't had the chance to confirm yet.

Reply:
I don't think it will. The reason we typically publish in json is because I didn't want to go through the hassle of creating a new avro schema for every module.

Reply:
You're right. Updated the serializer to `json`.